### ISSUE TITLE
Sorting in find seems to be ignored

### ISSUE THREAD

**Issue by 1024kilobyte** [OpenKitten/MongoKitten#52]
The "sortedBy" parameter seems to be ignored in findOne. The query itself is working and returning data, but always the the first in the database. Quering the database from the shell gives the correct result:

`
db.data.find({userToken: "ztJTb5kmODfBcbP-Em7n"}).sort({date: -1}).limit(1)
`

My code basically looks like:

```
let sortDocument: Document = ["date": -1]
let latestData = try dataCollection.findOne(matching: "userToken" == "ztJTb5kmODfBcbP-Em7n", sortedBy: sortDocument, projecting: nil, skipping: nil)
```

**Comment by Joannis** [comment]
Could you try `let sortDocument: Document = ["date": .int32(-1)]` ? It might be caused by your `-1` being an `Int64` instead of `Int32`.

**Comment by 1024kilobyte** [comment]
@Joannis: Thanks for your quick reply, unfortunately I did already try this with no effect...

**Comment by Joannis** [comment]
What MongoKitten version are you running?

**Comment by 1024kilobyte** [comment]
As I installed it yesterday it should be version 1.7.1 based on my SPM entry:
`.Package(url: "https://github.com/OpenKitten/MongoKitten.git", majorVersion: 1, minor: 7),`

**Comment by Joannis** [comment]
@1024kilobyte I'm sorry. I can't find any reason why it doesn't sort properly for you. I've looked through the code over and over again but nothing seems to be out of order, strangely enough.

**Comment by 1024kilobyte** [comment]
@Joannis: I finally found the reason for my problem and it was all my fault. The database server has been updated while running, which means only the shell has been updated. So in the Terminal I always saw:
`MongoDB shell version: 3.2.10`

Debugging through MongoKitten showed that the database had a maxWireVersion of 2. So I realized that the database was still on version 2.6.3. Stopping and restarting the db server fixed my problem.

Sorry for that and thanks for the support!
